src/box-node-sdk.ts

```diff
@@ -0,0 +1,463 @@
+	 * A basic session has no access to a user's refresh token. Because of this, once the session's tokens
+	 * expire the client cannot recover and a new session will need to be generated.
+	 *
+	 * @param {string} accessToken A user's Box API access token
+	 * @returns {BoxClient} Returns a new Box Client paired to a new BasicAPISession
+	 */
+	static getBasicClient(accessToken: string) {
+		return new BoxSDKNode({
+			clientID: '',
+			clientSecret: '',
+		}).getBasicClient(accessToken);
+	}
+
+	/**
+	 * Returns a Box Client with a persistent API session. A persistent API session helps manage the user's tokens,
+	 * and can refresh them automatically if the access token expires. If a central data-store is given, the session
+	 * can read & write tokens to it.
+	 *
+	 * @NOTE: If tokenInfo or tokenStore are formatted incorrectly, this method will throw an error. If you
+	 * haven't explicitly created either of these objects or are otherwise not completly confident in their validity,
+	 * you should wrap your call to getPersistentClient in a try-catch to handle any potential errors.
+	 *
+	 * @param {TokenInfo} tokenInfo A tokenInfo object to use for authentication
+	 * @param {TokenStore} [tokenStore] An optional token store for reading/writing tokens to session
+	 * @returns {BoxClient} Returns a new Box Client paired to a new PersistentAPISession
+	 */
+	getPersistentClient(tokenInfo: any /* FIXME */, tokenStore?: TokenStore) {
+		var apiSession = new PersistentAPISession(
+			tokenInfo,
+			tokenStore,
+			this.config,
+			this.tokenManager
+		);
+		return new BoxClient(apiSession, this.config, this.requestManager);
+	}
+
+	/**
+	 * Returns A Box Client with an Anonymous API Session. An Anonymous API Session has access to an anonymous
+	 * client-credentials token, which isn't tied to any specific user. Because of this, the client will only
+	 * have access to endpoints that allow client-credential tokens. All Anonymous API Sessions share the
+	 * same tokens, which allows them to refresh them efficiently and reduce load on both the application and
+	 * the API.
+	 *
+	 * @returns {BoxClient} Returns a new Box Client paired to a AnonymousAPISession
+	 */
+	getAnonymousClient() {
+		return new BoxClient(
+			this.anonymousSession,
+			this.config,
+			this.requestManager
+		);
+	}
+
+	/**
+	 * Create a new client using App Auth for the given entity.  This allows either
+	 * managing App Users (as the enterprise) or performing operations as the App
+	 * Users themselves (as a user).
+	 *
+	 * @param {string} type The type of entity to operate as, "enterprise" or "user"
+	 * @param {string} id The Box ID of the entity to operate as
+	 * @param {TokenStore} [tokenStore] (Optional) the token store to use for caching tokens
+	 * @returns {BoxClient} A new client authorized as the app user or enterprise
+	 */
+	getAppAuthClient(type: string, id: string, tokenStore?: TokenStore) {
+		if (type === 'enterprise' && !id) {
```

Comment:
Based on this conditional, it looks like `id` is an optional param (even though the comment docs don't indicate it is optional)